master/static/srv/notebook-entrypoint.sh

```diff
@@ -4,6 +4,17 @@ set -e
 
 export PATH="/run/determined/pythonuserbase/bin:$PATH"
 
+# If HOME is not explicitly set for a container, libcontainer (Docker) will
+# try to guess it by reading /etc/password directly, which will not work with
+# our linss_determined plugin (or any user-defined NSS plugin in a container).
+# The default is "/", but HOME must be a writable location for distributed
+# training, so we try to query the user system for a valid HOME, or default to
+# the working directory otherwise.
+if [ "$HOME" = "/" ] ; then
+    HOME="$(getent passwd "$(whoami)" | cut -d: -f6)" || HOME="$WORKING_DIR"
```

Comment:
Uh-oh, I just realized that the fallback branch _still_ can't be taken, despite the set/export separation, because the pipeline throws away the exit code of all but the last command:
```
$ a="$(x | cat)"; echo $?
bash: x: command not found
0
$ a="$(x)"; echo $?
bash: x: command not found
127
```
I'll approve this PR anyway; I would be fine with either making the two instances the same and then fixing them separately or fixing the other one too in this PR.

I haven't thought about it too hard, but the first solution that comes to mind is doing the first assignment in a line by itself and then doing the second one if the result is empty.

Reply:
I'm pretty sure the most correct way to do this is with a `set -o pipefail`, so I just did that.  However, I did it inside the subshell so that the user's `startup-hook.sh` is not affected, as I think `set -o pipefail` is a bit extreme to set automatically.  I mean... I think it is a good idea for users to use, but I think it's unexpected to do it automatically.

End result:

```sh
HOME="$(set -o pipefail; getent passwd "$(whoami)" | cut -d: -f6)" || HOME="$WORKING_DIR"
```